UI/UX improvements for Receive view

Lordhacker756 opened the pull request:
# Description
- Fixed the stale error message still being present during new invoice creation
- This was done by modifying the `reset()` action in the LSPStore, by introducing a param that allows fine-grained control of either just resetting the error or the complete states in the Store


Relates to issue: **ZEUS-0000**
PFA The Screen Recording of the issue here (Before)
- https://zeusln.slack.com/archives/C06U4KXR8GK/p1743180461821719

PFA the recording after the fix (Notice the stale error is no longer being carried to the next screen)

https://github.com/user-attachments/assets/790f81ed-825b-4e21-b1d9-41f2ffdff844




This pull request is categorized as a:

- [ ] New feature
- [x] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance
- [x] Other

## Checklist
- [x] I’ve run `yarn run tsc` and made sure my code compiles correctly
- [x] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [x] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [ ] No, I’m a fool
- [ ] Yes
- [x] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [ ] Android
- [x] iOS

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] Embedded LND
- [x] LND (REST)
- [x] LND (Lightning Node Connect)
- [ ] Core Lightning (CLNRest)
- [ ] Nostr Wallet Connect
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the ZEUS [Transfix page](https://app.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `yarn` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

kaloudis commented on components/Switch.tsx:
```diff
@@ -16,8 +16,8 @@ function Switch(props: SwitchProps) {
-                false: themeColor('disabled'),
-                true: trackEnabledColor || themeColor('highlight')
+                true: themeColor('disabled'),
+                false: trackEnabledColor || themeColor('highlight')
```
Personally, I'm not sold on this track change, but perhaps others are in favor of it.

I would split up the LSP error clearing and the track change into two separate PRs

Lordhacker756 commented on stores/LSPStore.ts:
```diff
@@ -95,6 +95,14 @@ export default class LSPStore {
+        this.error_msg = '';
+        this.flow_error = false;
+        this.flow_error_msg = '';
+    };
```
Sure, I was in a dilemma whether to use reset() or not, since reset was also clearing out values beyond the error. 
I'll remove the clearError() action and call the reset() action instead ✅

kaloudis commented on views/Receive.tsx:
```diff
@@ -3065,7 +3065,14 @@ export default class Receive extends React.Component<
+                                                    }
                                                     onPress={() => {
+                                                        // Clear the old error messages (if present)
+                                                        LSPStore.reset();
```
this approach is acceptable

Lordhacker756 replied:
Sure, thanks for your input, I'll push the code with the above approach

shubhamkmr04 reviewed (commented):
Please update the description

Lordhacker756 commented:
@shubhamkmr04 Updated the description

shubhamkmr04 commented:
> @shubhamkmr04 Updated the description

can you also clear the commit structure here? i dont think we need this [commit](https://github.com/ZeusLN/zeus/pull/2937/commits/bd01ad689bd059e8924221df1a8a260e60c37e76)

Lordhacker756 commented:
@shubhamkmr04 
I've removed the commit you told and also added the `After` change screen recording since we have the `Before` part in the Slack message link that I have added

Thanks

shubhamkmr04 commented:
@Lordhacker756 please fix the checks here

Lordhacker756 commented:
@shubhamkmr04 
I see, these checks failed because of the commit we removed, that commit was lint fixes.
I'll fix the lint issues and push a commit for the same

shubhamkmr04 commented:
> @shubhamkmr04 I see, these checks failed because of the commit we removed, that commit was lint fixes. I'll fix the lint issues and push a commit for the same

We want all of the commits to pass the checks. you need to reset the previous commits

Lordhacker756 commented:
@shubhamkmr04 
I removed the lint fix commit, went back to the commit whose lint was failing, amended that commit to include the lint fix and then pushed it. Is this correct

shubhamkmr04 commented:
> @shubhamkmr04 I removed the lint fix commit, went back to the commit whose lint was failing, amended that commit to include the lint fix and then pushed it. Is this correct

yes

shubhamkmr04 reviewed (commented):
LGTM
